pkg/tests/main_test.go

```diff
@@ -62,31 +63,35 @@ func ensureCompiledTestProviders(wd string) error {
 	bin := filepath.Join(wd, "..", "..", "bin")
 
 	type testProvider struct {
-		name        string
-		source      string
-		tfgenSource string
+		name             string
+		source           string
+		tfgenSource      string
+		expectTfgenError *string
 	}
 
+	internalErrorMsg := "Internal validation of the provider failed!"
```

Comment:
```suggestion
	const internalErrorMsg = "Internal validation of the provider failed!"
```

Reply:
this is not valid, as we can't then take the address of the constant, I've reverted it.